core/client/fs/src/main/java/alluxio/client/file/AlluxioFileInStream.java

```diff
@@ -399,10 +399,18 @@ private void triggerAsyncCaching(BlockInStream stream) {
       if (cache && (mLastBlockIdCached != blockId)) {
         // Construct the async cache request
         long blockLength = mOptions.getBlockInfo(blockId).getLength();
+        String host = dataSource.getHost();
+        // ALLUXIO-11172: If the worker is in a container, use the container hostname
```

Comment:
You may just leave the link as `https://github.com/Alluxio/alluxio/issues/11172` or `issues#11172`
Preferably, don't reuse #11172, but create a new issue.